Silicon/IdavillePkg/Library/FirmwareUpdateLib/FirmwareUpdateLib.c

```diff
@@ -257,6 +257,10 @@ GetFirmwareUpdateInfo (
     Status = GetComponentInfoByPartition ((UINT32)ImageHdr->UpdateHardwareInstance, IsBackup, &CompBase, &CompSize);
     if (EFI_ERROR(Status)) {
       DEBUG ((DEBUG_INFO, "No SBL component found !"));
+      // Free the UpdatePartition variable before returning
+      if (UpdatePartition != NULL) {
+          FreePool(UpdatePartition);
+      }
```

Comment:
It looks UpdatePartition check is not required. If UpdatePartition is NULL, it would return in line 245. 
Maybe also free UpdatePartition in below other error conditions (e.g. line 269).

Reply:
Updated accordinglt.